File: payment/src/main/java/com/yas/payment/model/Payment.java
package com.yas.payment.model;

import com.yas.commonlibrary.model.AbstractAuditEntity;
import com.yas.payment.model.enumeration.PaymentMethod;
import com.yas.payment.model.enumeration.PaymentStatus;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import java.math.BigDecimal;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "payment")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Payment extends AbstractAuditEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long orderId;

    private String checkoutId;

    private BigDecimal amount;

    private BigDecimal paymentFee;

    @Enumerated(EnumType.STRING)
    private PaymentMethod paymentMethod;

    @Enumerated(EnumType.STRING)
    private PaymentStatus paymentStatus;

    private String gatewayTransactionId;

    private String failureMessage;

    @SuppressWarnings("unused")
    private String paymentProviderCheckoutId;

}

File: storefront/modules/payment/models/Payment.ts
import { EPaymentMethod } from '../../common/models/EPaymentMethod';
import { EPaymentStatus } from './EPaymentStatus';

export type Payment = {
  id?: number;
  orderId?: number;
  checkoutId: string;
  amount: number;
  paymentFee?: number;
  paymentMethod?: EPaymentMethod;
  paymentStatus?: EPaymentStatus;
  gatewayTransactionId?: string;
  failureMessage?: string;
  paymentProviderCheckoutId?: string;
  attributes?: string;
};
